usage.go

```diff
@@ -0,0 +1,251 @@
+		if name != "" && name != "string" {
+			return name
+		}
+		return "String"
+	case reflect.Bool:
+		name := t.Name()
+		if name != "" && name != "bool" {
+			return name
+		}
+		return "Boolean"
+	case reflect.Int:
+		name := t.Name()
+		if name != "" && name != "int" {
+			return name
+		}
+		return "Integer"
+	case reflect.Int8:
+		name := t.Name()
+		if name != "" && name != "int8" {
+			return name
+		}
+		return "Integer(8 bits)"
+	case reflect.Int16:
+		name := t.Name()
+		if name != "" && name != "int16" {
+			return name
+		}
+		return "Integer(16 bits)"
+	case reflect.Int32:
+		name := t.Name()
+		if name != "" && name != "int32" {
+			return name
+		}
+		return "Integer(32 bits}"
+	case reflect.Int64:
+		name := t.Name()
+		if name != "" && name != "int64" {
+			return name
+		}
+		return "Integer(64 bits)"
+	case reflect.Uint:
+		name := t.Name()
+		if name != "" && name != "uint" {
+			return name
+		}
+		return "Unsigned Integer"
+	case reflect.Uint8:
+		name := t.Name()
+		if name != "" && name != "uint8" {
+			return name
+		}
+		return "Unsigned Integer(8 bits)"
+	case reflect.Uint16:
+		name := t.Name()
+		if name != "" && name != "uint16" {
+			return name
+		}
+		return "Unsigned Integer(16 bits)"
+	case reflect.Uint32:
+		name := t.Name()
+		if name != "" && name != "uint32" {
+			return name
+		}
+		return "Unsigned Integer(32 bits}"
+	case reflect.Uint64:
+		name := t.Name()
+		if name != "" && name != "uint64" {
+			return name
+		}
+		return "Unsigned Integer(64 bits)"
+	case reflect.Float32:
+		name := t.Name()
+		if name != "" && name != "float32" {
+			return name
+		}
+		return "Float"
+	case reflect.Float64:
+		name := t.Name()
+		if name != "" && name != "float64" {
+			return name
+		}
+		return "Float(64 bits)"
+	case reflect.Complex64:
+		return "Complex(64 bits)"
+	case reflect.Complex128:
+		return "Complex(128 bits)"
+	}
+	return fmt.Sprintf("%+v", t)
+}
+
+// Usage writes usage information to stderr using the default header and table format
+func Usage(prefix string, spec interface{}) error {
+	return Usagef(prefix, spec, os.Stderr, DefaultHeader, DefaultTableFormat)
+}
+
+// Usagef writes usage information to the specified io.Writer using the specifed header and usage format
+func Usagef(prefix string, spec interface{}, out io.Writer, header string, format string) error {
+
+	// gather first
+	infos, err := GatherInfo(prefix, spec)
+	if err != nil {
+		return err
+	}
+
+	if header != NoHeader {
+		headerTmpl, err := template.New("envconfig_header").Parse(header)
+		if err != nil {
+			return err
+		}
+		err = headerTmpl.Execute(out, path.Base(os.Args[0]))
+		if err != nil {
+			return err
+		}
+	}
+
+	var tabs *tabwriter.Writer = nil
+	var tmpl *template.Template
+	var tmplSpec = format
+	var usage usageInfo
+
+	// If format is prefixed with "table" then strip it off to get the per-line template, display the table headers,
+	// and inject a tab write filter
+	if strings.HasPrefix(format, "table") {
```

Comment:
The mini-DSL was modeled after what `docker` format does with its commands. It would be nice if the go template could support tables without having to introduce a tabwriter. 

removed the mini-DSL.